src/progressbar.c

```diff
@@ -150,12 +153,26 @@ void p_iter_destroy(p_iter* it) {
  */
 long p_iter_next(p_iter* it) {
     print_bar(it);
-    if (it->data && it->unit_size)
-        return it->iterations < it->length ? *(it->data+(it->iterations++ * it->unit_size)) : NULL;
-    else
-        return it->iterations < it->length ? it->iterations++ : NULL;
+    if (it->data && it->unit_size) {
+        // When iterating over data
+        if (it->iterations < it->length) {
+            // Perform pointer arithmetic to get the next element, cast it appropriately
+            long nextElement = *((long *)(it->data + (it->iterations++ * it->unit_size)));
+            return nextElement;
+        } else {
+            // End of iteration
+            return -1; // Special value indicating end
+        }
+    } else {
+        // When simply counting iterations
+        if (it->iterations < it->length) {
+            return it->iterations++;
+        } else {
+            // End of iteration
+            return -1; // Special value indicating end
+        }
+    }
```

Comment:
Noooooo my oneliner

Reply:
this change is sadly needed :(. Otherwise compile warnings won't go away.